### ISSUE TITLE
Mock Request Response Body Formatting and Validation

### ISSUE THREAD

**Issue by BenedictP** [httptoolkit/feedback#87]
Hi,

when I mock a request and I'm editing the response body it would be nice to have something like a formatting button and a validation if my input is a valid json/xml/etc.

---

_Does this affect you too? Click the +:smiley: icon in the top right and add a :+1: on issues to help decide where HTTP Toolkit goes next, or [go take a look & vote on the most popular ideas so far](https://github.com/httptoolkit/feedback/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc)._

**Comment by pimterry** [comment]
Do you mean when breakpointing a request and editing the body, or when setting fixed response body for a rule? I can see it being useful in both cases, just useful to know if one is particularly important.

There should already be validation I think, as long as the editor formatting is set to the right language (e.g. the drop down is set to JSON). Is that not working for you?

**Comment by BenedictP** [comment]
> Do you mean when breakpointing a request and editing the body, or when setting fixed response body for a rule? I can see it being useful in both cases, just useful to know if one is particularly important.

In my case: When I'm setting a fixed response body. But as you said for breakpointing a request it could be also useful.

> There should already be validation I think, as long as the editor formatting is set to the right language (e.g. the drop down is set to JSON). Is that not working for you?

I was editing a XML File and it works suboptimal:
<img width="739" alt="xml_example" src="https://user-images.githubusercontent.com/5817002/87023607-77283d80-c1c7-11ea-9e02-0ef6592b878e.png">
The line is not syntax highlighted but I would prefer it if it were underlined in red.

**Comment by pimterry** [comment]
This is now implemented: there's a format button available for both mock rules and breakpointed request/response bodies. Enjoy!

On XML validation, it seems that the editor I'm using internally (Monaco, from VS code) doesn't support XML validation by default, only syntax highlighting. I've filed a separate issue for that here: https://github.com/httptoolkit/feedback/issues/94. Feel free to :+1: vote on that there if that's important to you. It is a fair bit more work than these format buttons unfortunately, unless Monaco add built-in support for it, but I'm happy to look into adding custom support if it's a popular request.
